qdrant_client/async_qdrant_remote.py

```diff
@@ -727,6 +727,152 @@ async def recommend_groups(
             assert result is not None, "Recommend points API returned None"
             return result
 
+    async def discover(
+        self,
+        collection_name: str,
+        target: Optional[types.TargetVector] = None,
+        context: Optional[Sequence[types.ContextExamplePair]] = None,
+        query_filter: Optional[types.Filter] = None,
+        search_params: Optional[types.SearchParams] = None,
+        limit: int = 10,
+        offset: int = 0,
+        with_payload: Union[bool, List[str], types.PayloadSelector] = True,
+        with_vectors: Union[bool, List[str]] = False,
+        using: Optional[str] = None,
+        lookup_from: Optional[types.LookupLocation] = None,
+        consistency: Optional[types.ReadConsistency] = None,
+        timeout: Optional[int] = None,
+        **kwargs: Any,
+    ) -> List[types.ScoredPoint]:
+        if context is None:
+            context = []
+        if self._prefer_grpc:
+            target = (
+                RestToGrpc.convert_target_vector(target)
+                if target is not None
+                and isinstance(target, get_args_subscribed(models.RecommendExample))
+                else target
+            )
```

Comment:
does not work for me

script to reproduce
```python
from qdrant_client import QdrantClient
from qdrant_client.models import models as models
from qdrant_client import grpc as q_grpc
import numpy as np
import random

seed = 42
random.seed(seed)
np.random.seed(seed)

collection_name = "discovery_collection"
num_vectors = 10_000
vector_dim = 384


if __name__ == '__main__':
    client = QdrantClient(prefer_grpc=True)
    print(f"Recreating collection {collection_name}")
    client.recreate_collection(
        collection_name=collection_name,
        vectors_config=models.VectorParams(size=vector_dim, distance=models.Distance.COSINE),
    )
    vectors = np.random.rand(num_vectors, vector_dim).tolist()
    print("Uploading vectors")
    client.upload_collection(collection_name=collection_name, vectors=vectors, parallel=6)
    print("Vectors have been uploaded")
    client.discover(collection_name=collection_name, target=1)
```

Traceback:
```
grpc._channel._InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
	status = StatusCode.INTERNAL
	details = "failed to decode Protobuf message: PointId.point_id_options: VectorExample.example: DiscoverPoints.target: invalid wire type: LengthDelimited (expected Varint)"
	debug_error_string = "UNKNOWN:Error received from peer  {created_time:"2023-11-17T16:41:49.569981+01:00", grpc_status:13, grpc_message:"failed to decode Protobuf message: PointId.point_id_options: VectorExample.example: DiscoverPoints.target: invalid wire type: LengthDelimited (expected Varint)"}"
```

Reply:
I used this build `dc3667b66ac4`

Reply:
Unfortunately the dev image with the latest proto changes hadn't been uploaded when you tried it. Can you try again? It works fine with `b2fe2f9ecdf2`.
I also updated the tests for running them with the grpc client too.